lib/rules/jsx-no-literals.js

```diff
@@ -65,56 +65,10 @@ const commonPropertiesSchema = {
   },
 };
 
-/**
- * @typedef RawElementConfigProperties
- * @property {boolean} [noStrings]
- * @property {string[]} [allowedStrings]
- * @property {boolean} [ignoreProps]
- * @property {boolean} [noAttributeStrings]
- *
- * @typedef RawOverrideConfigProperties
- * @property {boolean} [allowElement]
- * @property {boolean} [applyToNestedElements=true]
- *
- * @typedef {RawElementConfigProperties} RawElementConfig
- * @typedef {RawElementConfigProperties & RawElementConfigProperties} RawOverrideConfig
- *
- * @typedef RawElementOverrides
- * @property {Record<string, RawOverrideConfig>} [elementOverrides]
- *
- * @typedef {RawElementConfig & RawElementOverrides} RawConfig
- *
- * ----------------------------------------------------------------------
- *
- * @typedef ElementConfigType
- * @property {'element'} type
- *
- * @typedef ElementConfigProperties
- * @property {boolean} noStrings
- * @property {Set<string>} allowedStrings
- * @property {boolean} ignoreProps
- * @property {boolean} noAttributeStrings
- *
- * @typedef OverrideConfigProperties
- * @property {'override'} type
- * @property {string} name
- * @property {boolean} allowElement
- * @property {boolean} applyToNestedElements
- *
- * @typedef {ElementConfigType & ElementConfigProperties} ElementConfig
- * @typedef {OverrideConfigProperties & ElementConfigProperties} OverrideConfig
- *
- * @typedef ElementOverrides
- * @property {Record<string, OverrideConfig>} elementOverrides
- *
- * @typedef {ElementConfig & ElementOverrides} Config
- * @typedef {Config | OverrideConfig} ResolvedConfig
- */
-
 /**
  * Normalizes the element portion of the config
- * @param {RawConfig} config
- * @returns {ElementConfig}
+ * @param {import("../../types/rules/jsx-no-literals").RawConfig} config
+ * @returns {import("../../types/rules/jsx-no-literals").ElementConfig}
```

Comment:
After adding `/** @type {import('eslint').Rule.RuleModule} */` to `lib/rules/jsx-no-literals.js`, the generated `lib/rules/jsx-no-literals.d.ts` has the following content:

```ts
declare const _exports: import('eslint').Rule.RuleModule;
export = _exports;
export type RawElementConfigProperties = {
    noStrings?: boolean;
    allowedStrings?: string[];
    ignoreProps?: boolean;
    noAttributeStrings?: boolean;
};
```

TypeScript will throw error when using this .d.ts file: `An export assignment cannot be used in a module with other exported elements.ts(2309)`

Although types like `RawElementConfigProperties` are only used internally, using JSDoc `@typedef` would export them (see issue https://github.com/microsoft/TypeScript/issues/46011) and cause the error above.

To resolve the issue, I added a new `types/rules/jsx-no-literals.d.ts` file and moved all types defined with `@typedef` into it.